CMakeLists.txt

```diff
@@ -246,7 +246,6 @@ list(APPEND RUN_CLANG_TIDY_BIN_ARGS
     -clang-apply-replacements-binary ${CLANG_AR_BIN}
     -style=file
     -quiet
-    -header-filter="${CMAKE_SOURCE_DIR}/[^b].*"
```

Comment:
It's true that it could potentially match with other dirs staring with `b`. At the time I added this there was no better way to exclude the `build` dir without expanding the pattern exponentially in the number of letters of the word `build`. I did this because we were getting failures. If it works for now so be it, I'm just concerned we may run into problems in the future.

Reply:
As far as I can tell its not doing anything on my machine or CI - so perhaps we continue with it removed and only add back if this becomes an issue again? 

It might only become an issue with in-tree builds which I don't think we should be doing anyways?

Reply:
I suppose it's possible that your `NOLINT` directives for `misc-include-cleaner` may be resulting in a similar behavior. Our build is out-of-tree because we don't produce object files or binaries in the same directory as source files, but in the past the build folder was getting caught by the file globs being sent to clang-tidy. Anyway, happy to get rid of it if it's not doing anything. 👍

Reply:
I remember a bit more context now. Clang tidy was warning because it was checking the files included using the `#include` directives, as it normally does. (Only cpp files are fed to tidy, and it checks the headers that are included in them in this way. So the filter just ensures that tidy doesn't warn on those files matching it.